plugins/history/databasemanager.cpp

```diff
@@ -0,0 +1,140 @@
+		if (!db.open()) {
+			//Database not open
+			return;
+		}
+
+		//If the messages table does not exist, lets create it.
+		db.exec(
+					"CREATE TABLE IF NOT EXISTS \"messages\" ("
+                    "\"" + DatabaseConstants::columnId() + "\" Integer Primary Key Autoincrement Not Null, "
+                    "\"" + DatabaseConstants::columnTimeStamp() + "\" Integer, "
+                    "\"" + DatabaseConstants::columnMessage() + "\" Text, "
+                    "\"" + DatabaseConstants::columnProtocol() + "\" Text Not Null, "
+                    "\"" + DatabaseConstants::columnAccount() + "\" Text Not Null, "
+                    "\"" + DatabaseConstants::columnDirection() + "\" Integer Not Null, "
+                    "\"" + DatabaseConstants::columnImportance() + "\" Integer, "
+                    "\"" + DatabaseConstants::columnContact() + "\" Text, "
+                    "\"" + DatabaseConstants::columnSubject() + "\" Text, "
+                    "\"" + DatabaseConstants::columnSession() + "\" Text, "
+                    "\"" + DatabaseConstants::columnSessionName() + "\" Text, "
+                    "\"" + DatabaseConstants::columnFrom() + "\" Text, "
+                    "\"" + DatabaseConstants::columnFromName() + "\" Text, "
+                    "\"" + DatabaseConstants::columnTo() + "\" Text, "
+                    "\"" + DatabaseConstants::columnToName() + "\" Text, "
+                    "\"" + DatabaseConstants::columnState() + "\" Integer, "
+                    "\"" + DatabaseConstants::columnType() + "\" Integer, "
+                    "\"" + DatabaseConstants::columnIsGroup() + "\" Integer Default'0') "
+					);
+	}
+}
+
+void DatabaseManager::insertMessage(Kopete::Message &message)
+{
+	QSqlQuery query(db);
+
+	//Prepare an SQL query to insert the message to the db
+	query.prepare("INSERT INTO `messages` (`timestamp`, `message`, `account`, `protocol`, `direction`, `importance`, `contact`, `subject`, "
+		      " `session`, `session_name`, `from`, `from_name`, `to`, `to_name`, `state`, `type`, `is_group`) "
+		      " VALUES (:timestamp, :message, :account, :protocol, :direction, :importance, :contact, :subject, :session, :session_name, "
+		      " :from, :from_name, :to, :to_name, :state, :type, :is_group)");
+
+	//Add the values to the database fields.
+    query.bindValue(":" + DatabaseConstants::columnTimeStamp() , QString::number(message.timestamp().toTime_t()));
+    query.bindValue(":" + DatabaseConstants::columnMessage(), message.parsedBody());
+    query.bindValue(":" + DatabaseConstants::columnAccount(), message.manager()->account()->accountId());
+    query.bindValue(":" + DatabaseConstants::columnProtocol(), message.manager()->account()->protocol()->pluginId());
+    query.bindValue(":" + DatabaseConstants::columnDirection(), QString::number(message.direction()));
+    query.bindValue(":" + DatabaseConstants::columnImportance(), QString::number(message.importance()));
```

Comment:
I think it would be a good idea to create a private function that takes the `key`, `value` and binds it to `query`.